src/zc/form/field.py

```diff
@@ -390,6 +401,17 @@ def _validate(self, value):
                 f.validate(v)
         super(Combination, self)._validate(value)
 
+    def bind(self, object):
+        clone = super(Combination, self).bind(object)
+        # We need to bind the fields too, e.g. for Choice fields
+        clone_fields = []
+        for field in clone.fields:
+            clone_fields.append(field.bind(object))
+        clone_fields = tuple(clone_fields)
+        clone.fields = clone_fields
```

Comment:
It could even be shortened to `clone.fields = tuple(field.bind(object) for field in clone.fields)` aka without the square brackets. I think this is more readable than the current code.